core/src/main/java/org/hisp/dhis/android/core/category/CategoryCategoryOptionLinkStoreImpl.java

```diff
@@ -4,47 +4,126 @@
     private final SQLiteStatement insertStatement;
+    private final SQLiteStatement deleteStatement;
+    private final SQLiteStatement updateStatement;
 
     private static final String INSERT_STATEMENT =
             "INSERT INTO " + CategoryCategoryOptionLinkModel.TABLE + " (" +
                     CategoryCategoryOptionLinkModel.Columns.CATEGORY + ", " +
                     CategoryCategoryOptionLinkModel.Columns.CATEGORY_OPTION + ") " +
                     "VALUES(?, ?);";
 
+    private static final String DELETE_STATEMENT =
+            "DELETE FROM " + CategoryCategoryOptionLinkModel.TABLE +
+                    " WHERE " + CategoryCategoryOptionLinkModel.Columns.CATEGORY + " =?" + " AND "
+                    + CategoryCategoryOptionLinkModel.Columns.CATEGORY_OPTION + "=?;";
+
+    private static final String UPDATE_STATEMENT =
+            "UPDATE " + CategoryCategoryOptionLinkModel.TABLE + " SET " +
+                    CategoryCategoryOptionLinkModel.Columns.CATEGORY + " =?, " +
+                    CategoryCategoryOptionLinkModel.Columns.CATEGORY_OPTION + " =?" +
+                    " WHERE " + CategoryCategoryOptionLinkModel.Columns.CATEGORY + " =? AND " +
+                    CategoryCategoryOptionLinkModel.Columns.CATEGORY_OPTION + " =?;";
+
+    private static final String FIELDS =
+            CategoryCategoryOptionLinkModel.TABLE + "."
+                    + CategoryCategoryOptionLinkModel.Columns.CATEGORY + "," +
+                    CategoryCategoryOptionLinkModel.TABLE + "."
+                    + CategoryCategoryOptionLinkModel.Columns.CATEGORY_OPTION;
+
+
+    private static final String QUERY_ALL_CATEGORY_OPTION_LINKS = "SELECT " +
+            FIELDS + " FROM " + CategoryCategoryOptionLinkModel.TABLE;
 
     public CategoryCategoryOptionLinkStoreImpl(DatabaseAdapter databaseAdapter) {
         this.databaseAdapter = databaseAdapter;
         this.insertStatement = databaseAdapter.compileStatement(INSERT_STATEMENT);
+        this.deleteStatement = databaseAdapter.compileStatement(DELETE_STATEMENT);
+        this.updateStatement = databaseAdapter.compileStatement(UPDATE_STATEMENT);
     }
 
     @Override
-    public long insert(@NonNull CategoryCategoryOptionLinkModel entity) {
+    public long insert(@NonNull CategoryCategoryOptionLinkModel categoryOptionLinkModel) {
 
-        validate(entity);
+        validate(categoryOptionLinkModel);
 
-        bind(insertStatement, entity);
+        bind(insertStatement, categoryOptionLinkModel);
 
         return executeInsert();
     }
 
+    @Override
+    public int delete() {
+        return databaseAdapter.delete(CategoryCategoryOptionLinkModel.TABLE);
+    }
+
+    @Override
+    public boolean delete(@NonNull CategoryCategoryOptionLinkModel categoryOptionLinkModel) {
+        validate(categoryOptionLinkModel);
+
+        bind(deleteStatement, categoryOptionLinkModel);
+
+        return execute(deleteStatement);
+    }
+
+    @Override
+    public boolean update(
```

Comment:
To update one row only is necessary one model, you are pass as argument new and old, can you refactor this method?

Reply:
The problem with removing the extra parameter is that this table only has two columns(category and categoryOption) and at the same time they are the primary keys to this table for this reason, if I remove the second argument, I'll be forced to add two new parameters (category and categoryOption) as the where clause to identify which object, the client of the method wants to update, for example:

Imagine that I have this row on the table **CategoryCategoryOptionLink**

| category        | categoryOption |
| ------------- |:-------------:| 
| id_category_1      | id_categoryOption_1 |

And I want to update it to:

**CategoryCategoryOptionLink table**

| category        | categoryOption |
| ------------- |:-------------:| 
| id_category_2      | id_categoryOption_2 |

With the actual signature of the method I just pass:

**OLD (the where clause)**
``` 
 {
    "category": "category1", 
    "categoryOption":"categoryOption1"
}
```
**NEW (the data to update)**
``` 
 {
    "category": "category2", 
    "categoryOption":"categoryOption2"
}
``` 
With only one model, I'll lose the where clause.

Reply:
ok is true